pkg/nfd-master/nfd-master.go

```diff
@@ -605,20 +604,85 @@ func (m *nfdMaster) nfdAPIUpdateOneNode(nodeName string) error {
 	return nil
 }
 
-func (m *nfdMaster) refreshNodeFeatures(cli *kubernetes.Clientset, nodeName string, annotations, labels map[string]string, features *nfdv1alpha1.Features) error {
+// filterExtendedResources filters extended resources and returns a map
+// of valid extended resources.
+func (m *nfdMaster) filterExtendedResources(features *nfdv1alpha1.Features, extendedResources ExtendedResources) ExtendedResources {
+	outExtendedResources := ExtendedResources{}
+	deniedNs := map[string]struct{}{"kubernetes.io": {}}
+	deniedWildCarNs := map[string]struct{}{".kubernetes.io": {}}
+	for extendedResource, capacity := range extendedResources {
+		// Add possibly missing default ns
+		extendedResource = addNs(extendedResource, nfdv1alpha1.ExtendedResourceNs)
+		// Check if given NS is allowed
+		ns, _ := splitNs(extendedResource)
+		if isNamespaceDenied(ns, deniedWildCarNs, deniedNs) {
+			if ns != nfdv1alpha1.ExtendedResourceNs && ns != nfdv1alpha1.FeatureLabelNs &&
+				!strings.HasSuffix(ns, nfdv1alpha1.ExtendedResourceSubNsSuffix) && !strings.HasSuffix(ns, nfdv1alpha1.FeatureLabelSubNsSuffix) {
+				klog.Errorf("Namespace %q is not allowed. Ignoring Extended Resource  %q\n", ns, extendedResource)
```

Comment:
You don't need the linefeed. Also, just remembered that we've tried to have a convention of starting log messages with lower case letter...

```suggestion
				klog.Errorf("namespace %q is not allowed. Ignoring Extended Resource  %q", ns, extendedResource)
```